amethyst_core/src/system_ext.rs

```diff
@@ -0,0 +1,113 @@
+//! Extension system utilities.
+//!
+//! This modules contains an extension trait for the System trait which adds useful transformation
+//! functions.
+
+use shred::{RunningTime, SystemData};
+use specs::prelude::{Read, System};
+
+/// Extension functionality associated systems.
+pub trait SystemExt {
+    /// Make a system pausable by tying it to a specific value of a resource.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate amethyst;
+    /// use amethyst::{
+    ///     ecs::{System, Write},
+    ///     shred::DispatcherBuilder,
+    ///     prelude::*,
+    /// };
+    ///
+    /// #[derive(PartialEq, Eq)]
+    /// enum CurrentState {
+    ///     Disabled,
+    ///     Enabled,
+    /// }
+    ///
+    /// impl Default for CurrentState {
```

Comment:
Non-default resources require `ReadExpect`, and have the "Tried to fetch a resource" error when they can't be found which is rather cryptic:

```
Tried to fetch a resource, but the resource does not exist.
Try adding the resource by inserting it manually or using the `setup` method.
You can get the type name of the missing resource by enabling `shred`'s `nightly` feature
```

This has generally tipped me more in favor of using resources which `Default`, since it guards against this kind of error in the type system.